### ISSUE TITLE
Define strategy for guides to adopt new feature versions

### ISSUE THREAD

**Issue by yeekangc** [OpenLiberty/guides-common#52]
We need to work out when to update all the sample apps/code assets to use new feature versions.  Annually?  Every release?

**Comment by evelinec** [comment]
I believe all guides now runs with the release build. Perhaps we had resolved the release part. Still need to think about the new feature versions.

**Comment by evelinec** [comment]
Looks like this could be part of the automation, where possible.

**Comment by proubatsis** [comment]
The first question that needs to be asked is what we want to achieve by upgrading the feature to a new version. Are we just upgrading it for the purpose of being up-to-date, or do we want to show off new features that may be incorporated? If it is the latter then it may be worthwhile to create a new guide depending on the context of the situation.

Typically, versioning takes the format of the major number representing a breaking change. If only the minor number changes, then it should be relatively simple to update as nothing should break as a result.

---

As far as the upgrade process goes, these would generally be the required steps:

1. Update `server.xml` to have the new feature version
1. Update `pom.xml/build.gradle`
1. Build & Test
1. If there are any breaking changes, adjust code wherever necessary
1. Commit & PR

With respect to automation, updating `pom.xml/build.gradle` might be tricky, and fixing any breaking changes would require manual changes.

---

@evelinec @yeekangc

**Comment by evelinec** [comment]
We'll put this issue on hold till further planning and design. Thanks.

**Comment by Kubik42** [comment]
@proubatsis @evelinec @yeekangc Upgrading minor versions doesn't seem that difficult. I would say that, just to test it out for now, we can run the version change manually via some script. This would pull the latest release and dev builds, then extract a list of all their features, and then filter this list to find what features have new versions. 

Once this is done, we look which of these version changes are major/minor. For minor, we upgrade and kick off builds to verify that nothing is broken. For major, we first verify that nothing new needs demonstration and them upgrade.

**Comment by evelinec** [comment]
All of the MP guides (16 of them) had been updated. There are 4 remaining ones that need to have install feature updates in the pom.xml. Then the automation can be tested and set forward to run on a schedule basics.

**Comment by evelinec** [comment]
The feature update automation has running regularly to check for updates.
